keeper/src/main/scala/zio/keeper/membership/swim/Nodes.scala

```diff
@@ -46,13 +46,13 @@ final class Nodes(
               .commit
               .tap(
                 _ => {
-                  val event = if (newState == NodeState.Healthy && prev == NodeState.Init) {
-                    Join(id)
+                  if (newState == NodeState.Healthy && prev == NodeState.Init) {
```

Comment:
This can be polished with ZIO.whenCaseM.

Reply:
I knew that you gonna say that :)